go/parameter_type_registry.go

```diff
@@ -74,7 +102,126 @@ func NewParameterTypeRegistry() *ParameterTypeRegistry {
 	if err != nil {
 		panic(err)
 	}
-	result.DefineParameterType(floatParameterType)
+	err = result.DefineParameterType(floatParameterType)
+	if err != nil {
+		panic(err)
+	}
+
+	doubleParameterType, err := NewParameterType(
+		"double",
+		FLOAT_REGEXPS,
+		"float64",
+		func(args ...*string) interface{} {
+			f, err := transformer.Transform(*args[0], reflect.Float64)
+			if err != nil {
+				panic(err)
+			}
+			return f
+		},
+		false,
+		false,
+		false,
+	)
+	if err != nil {
+		panic(err)
+	}
+	err = result.DefineParameterType(doubleParameterType)
+	if err != nil {
+		panic(err)
+	}
+
+	byteParameterType, err := NewParameterType(
+		"byte",
+		FLOAT_REGEXPS,
+		"int8",
+		func(args ...*string) interface{} {
+			f, err := transformer.Transform(*args[0], reflect.Int8)
+			if err != nil {
+				panic(err)
+			}
+			return f
+		},
+		false,
+		false,
+		false,
+	)
+	if err != nil {
+		panic(err)
+	}
+	err = result.DefineParameterType(byteParameterType)
+	if err != nil {
+		panic(err)
+	}
+
+	shortParameterType, err := NewParameterType(
+		"short",
+		FLOAT_REGEXPS,
+		"int16",
+		func(args ...*string) interface{} {
+			f, err := transformer.Transform(*args[0], reflect.Int16)
+			if err != nil {
+				panic(err)
+			}
+			return f
+		},
+		false,
+		false,
+		false,
+	)
+	if err != nil {
+		panic(err)
+	}
+	err = result.DefineParameterType(shortParameterType)
+	if err != nil {
+		panic(err)
+	}
+
+	longParameterType, err := NewParameterType(
+		"long",
+		FLOAT_REGEXPS,
+		"int32",
+		func(args ...*string) interface{} {
+			f, err := transformer.Transform(*args[0], reflect.Int64)
+			if err != nil {
+				panic(err)
+			}
+			return f
+		},
+		false,
+		false,
+		false,
+	)
+	if err != nil {
+		panic(err)
+	}
+	err = result.DefineParameterType(longParameterType)
+	if err != nil {
+		panic(err)
+	}
+
+	bigdecimalParameterType, err := NewParameterType(
+		"bigdecimal",
+		FLOAT_REGEXPS,
+		"BigFloat",
+		func(args ...*string) interface{} {
+			f, err := transformer.Transform(*args[0], BigFloatKind)
+			if err != nil {
+				panic(err)
+			}
+			return f
+		},
+		false,
+		false,
+		false,
+	)
+	if err != nil {
+		panic(err)
+	}
+	err = result.DefineParameterType(bigdecimalParameterType)
+	if err != nil {
+		panic(err)
+	}
+
```

Comment:
This looks like there is some pattern here
Is it possible to factorize it?

Maybe that would not be welcome or easy in go?

Reply:
I see what you mean but I think it would be clunky to refactor :-/

Reply:
true